examples/funcalign/searchlight_srm_example.py

```diff
@@ -0,0 +1,152 @@
+import sys
+import scipy.io as sio
+from scipy.stats import stats
+from brainiak.searchlight.searchlight import Searchlight
+from brainiak.funcalign.srm import SRM
+import warnings
+
+# parameters
+sl_rad = 1 #searchlight length (of each edge) will be 1+2*sl_rad
+max_blk_edge = 1 #won't change computational results, has effect on MPI processing
+nfeature = 10 #number of features in SRM for each searchlight
+n_iter = 10 #number of interations in SRM
+
+# sanity check
+if sl_rad <= 0 or max_blk_edge <= 0:
+  raise ValueError('sl_rad and max_blk_edge must be positive')
+if nfeature > (1+2*sl_rad)**3:
+  print ('nfeature truncated')
+  nfeature = int((1+2*sl_rad)**3)
+
+# MPI parameters, do not need to change
+comm = MPI.COMM_WORLD
+rank = comm.rank
+size = comm.size
+
+# load data
+movie_file = sio.loadmat('data/sl_movie_data.mat')
+movie_data = movie_file['data']
+
+# Dataset size parameters
+dim1,dim2,dim3,ntr,nsubj = movie_data.shape
+
+# preprocess data, zscore and set NaN to 0
+all_data = [] # first half train, second half test
+for s in range(nsubj):
+  with warnings.catch_warnings():
+    warnings.simplefilter("ignore")
+    train_tmp = np.nan_to_num(stats.zscore(movie_data[:,:,:,:int(ntr/2),s],axis=3,ddof=1))
+    test_tmp = np.nan_to_num(stats.zscore(movie_data[:,:,:,int(ntr/2):,s],axis=3,ddof=1))
+  all_data.append(np.concatenate((train_tmp,test_tmp),axis=3))
+
+# Generate mask: mask is a 3D binary array, with active voxels being 1. I simply set 
+# all voxels to be active in this example, but you should set the mask to fit your ROI
+# in practice.
+mask = np.ones((dim1,dim2,dim3), dtype=np.bool)
+
+# Create searchlight object
+sl = Searchlight(sl_rad=sl_rad, max_blk_edge=max_blk_edge)
+
+# Distribute data to processes
+# the first argument of "distribute" is a list of 4D arrays, and each 4D array is data 
+# from a single subject
+sl.distribute(all_data, mask)
+# broadcast something that should be shared by all ranks 
+sl.broadcast([n_iter,nfeature])
+
+# time segment matching experiment. Define your own experiment function here
+def timesegmentmatching_accuracy(data, win_size=6): 
+  nsubjs = len(data)
+  (ndim, nsample) = data[0].shape
+  accu = np.zeros(shape=nsubjs)
+  nseg = nsample - win_size 
+  # mysseg prediction prediction
+  trn_data = np.zeros((ndim*win_size, nseg))
+  # the trn data also include the tst data, but will be subtracted when 
+  # calculating A
+  for m in range(nsubjs):
+      for w in range(win_size):
+          trn_data[w*ndim:(w+1)*ndim,:] += data[m][:,w:(w+nseg)]
+  for tst_subj in range(nsubjs):
+    tst_data = np.zeros((ndim*win_size, nseg))
+    for w in range(win_size):
+      tst_data[w*ndim:(w+1)*ndim,:] = data[tst_subj][:,w:(w+nseg)]
+
+    A =  np.nan_to_num(stats.zscore((trn_data - tst_data),axis=0, ddof=1))
+    B =  np.nan_to_num(stats.zscore(tst_data,axis=0, ddof=1))
+    # normalize A and B
+    A = A/np.linalg.norm(A,axis=0)
+    B = B/np.linalg.norm(B,axis=0)
+    corr_mtx = B.T.dot(A)
```

Comment:
Could you use our optimized function?
http://pythonhosted.org/brainiak/brainiak.fcma.html#brainiak.fcma.util.compute_correlation